src/stream-analytics/azext_stream_analytics/_params.py

```diff
@@ -0,0 +1,180 @@
+        c.argument('name', options_list=['--name', '-n'], help='The name of the streaming job.')
+        c.argument('tags', tags_type)
+        c.argument('events_out_of_order_policy', arg_type=get_enum_type(EventsOutOfOrderPolicy),
+                   help='Indicates the policy to apply to events that arrive out of order in the input event stream.')
+        c.argument('output_error_policy', arg_type=get_enum_type(OutputErrorPolicy),
+                   help='Indicates the policy to apply to events that arrive at the output.')
+        c.argument('events_out_of_order_max_delay_in_seconds', type=int,
+                   help='The maximum tolerable delay in seconds where out-of-order events can be adjusted to be back in order.')
+        c.argument('events_late_arrival_max_delay_in_seconds', type=int,
+                   help='The maximum tolerable delay in seconds where events arriving late could be included.  Supported range is -1 to 1814399 (20.23:59:59 days) and -1 is used to specify wait indefinitely.')
+        c.argument('data_locale',
+                   help='The data locale of the stream analytics job.')
+
+    with self.argument_context('stream-analytics job delete') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('name', options_list=['--name', '-n'], help='The name of the streaming job.')
+
+    with self.argument_context('stream-analytics job show') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('name', options_list=['--name', '-n'], help='The name of the streaming job.')
+        c.argument('expand', action='store_true',
+                   help='Expand inputs, transformation, outputs and functions of the streaming job')
+
+    with self.argument_context('stream-analytics job list') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('expand', action='store_true',
+                   help='Expand inputs, transformation, outputs and functions of the streaming job')
+
+    with self.argument_context('stream-analytics job start') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('name', options_list=['--name', '-n'], help='The name of the streaming job.')
+        c.argument('output_start_mode', arg_type=get_enum_type(OutputStartMode),
+                   help='Output start mode')
+        c.argument('output_start_time',
+                   help='Output start time, must have a value if --output-start-mode is set to CustomTime.')
+
+    with self.argument_context('stream-analytics job stop') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('name', options_list=['--name', '-n'], help='The name of the streaming job.')
+
+    with self.argument_context('stream-analytics input create') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
```

Comment:
should be job, and supports both name and resource id right?

Reply:
yes, id_part added, support both id and name now.